editors/code/src/commands.ts

```diff
@@ -260,6 +261,33 @@ export function openCargoToml(ctx: CtxInit): Cmd {
     };
 }
 
+export function revealDependency(ctx: CtxInit): Cmd {
+    return async (editor: RustEditor) => {
+        const rootPath = vscode.workspace.workspaceFolders![0].uri.fsPath;
+        const documentPath = editor.document.uri.fsPath;
+        if (documentPath.startsWith(rootPath)) return;
+        const dep = ctx.dependencies?.getDependency(documentPath);
+        if (dep) {
+            await ctx.treeView?.reveal(dep, { select: true, expand: true });
+        } else {
+            let documentPath = editor.document.uri.fsPath;
+            const parentChain: DependencyId[] = [{ id: documentPath.toLowerCase() }];
+            do {
+                documentPath = path.dirname(documentPath);
+                parentChain.push({ id: documentPath.toLowerCase() });
+            } while (!ctx.dependencies?.contains(documentPath));
+            parentChain.reverse();
+            for (const idx in parentChain) {
+                await ctx.treeView?.reveal(parentChain[idx], { select: true, expand: true });
```

Comment:
do you mind adding a comment explaining why this bit is necessary? is it to show a dependency of a dependency?

Reply:
I'll add it.
Its because the TreeView is Lazy, so at first it only has the root dependencies: Example:
- core
- alloc
- std

if I want to reveal alloc/src/str.rs, I have to:
1. reveal every children of alloc
- core
- alloc
&emsp;|-beches
&emsp;|-src
&emsp;|- ...
- std
2. reveal every children of src:
- core
- alloc
&emsp;|-beches
&emsp;|-src
&emsp;&emsp;| - lib.rs
&emsp;&emsp;| - str.rs    <------- FOUND IT!
&emsp;&emsp;| - ...
&emsp;|- ...
- std

So basically, i have to calculate the parents of the file that I want to reveal until i find the root crate path.